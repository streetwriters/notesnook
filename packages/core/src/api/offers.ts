/*
This file is part of the Notesnook project (https://notesnook.com/)

Copyright (C) 2023 Streetwriters (Private) Limited

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

import { CLIENT_ID } from "../common.js";
import hosts from "../utils/constants.js";
import http from "../utils/http.js";

export class Offers {
  static async getCode(promo: string, platform: "ios" | "android" | "web") {
    const result = await http.get(
      `${hosts.SUBSCRIPTIONS_HOST}/offers?promoCode=${promo}&clientId=${CLIENT_ID}&platformId=${platform}`
    );
    return result.code;
  }
}
