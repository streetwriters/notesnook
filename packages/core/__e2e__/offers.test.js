/*
This file is part of the Notesnook project (https://notesnook.com/)

Copyright (C) 2023 Streetwriters (Private) Limited

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

import hosts from "../src/utils/constants.ts";
import { Offers } from "../src/api/offers.ts";
import { test, expect } from "vitest";

test("get offer code", async () => {
  hosts.SUBSCRIPTIONS_HOST = "https://subscriptions.streetwriters.co";
  expect(await Offers.getCode("TESTOFFER", "android")).toMatchSnapshot(
    "offer-code"
  );
});

test("get invalid offer code", async () => {
  hosts.SUBSCRIPTIONS_HOST = "https://subscriptions.streetwriters.co";
  await expect(() => Offers.getCode("INVALIDOFFER", "android")).rejects.toThrow(
    /Not found/i
  );
});
