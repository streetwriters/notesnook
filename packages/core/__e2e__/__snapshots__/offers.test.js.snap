// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`get offer code > offer-code 1`] = `"123"`;
